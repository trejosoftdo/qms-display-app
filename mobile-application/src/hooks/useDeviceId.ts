import { useEffect, useState } from "react";
import { getDeviceId } from "../common/device-connection";
import { useProgress } from "../common/hooks";
import { DeviceData, Progress } from "../common/models";

/**
 * Hook to get the device identifier
 * @returns Progress<DeviceData>
 */
const useDeviceId = (): Progress<DeviceData> => useProgress<DeviceData>(() => getDeviceId().then((deviceId) => ({ deviceId })));

export default useDeviceId;
