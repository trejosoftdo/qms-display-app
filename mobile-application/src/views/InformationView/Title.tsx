import * as React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { List } from 'react-native-paper';
import { useTranslation } from 'react-i18next';
import { TRANSLATION_DETAILS_KEY } from '../../common/translations/translation-keys';

/**
 * TitleProps defines the props for the Title Component.
 */
interface TitleProps {}

/**
 * A component to show a title
 *
 * @param {TitleProps} props - The props for the Title component.
 */
const Title: React.FC<TitleProps> = (props: TitleProps) => {
  const { t } = useTranslation();
  return (
    <View style={styles.title}>
      <List.Icon {...props} icon="information" />
      <Text style={styles.titleText}>{t(TRANSLATION_DETAILS_KEY)}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  title: {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'row',
    width: '50%',
  },
  titleText: {
    paddingLeft: 10,
    fontSize: 16
  },
});

export default Title;