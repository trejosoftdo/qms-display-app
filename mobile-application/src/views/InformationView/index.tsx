import * as React from 'react';
import { StyleSheet } from 'react-native';
import { List } from 'react-native-paper';
import { useTranslation } from 'react-i18next';
import useDeviceId from '../../hooks/useDeviceId';
import { TRANSLATION_ID_KEY, TRANSLATION_INFORMATION_KEY } from '../../common/translations/translation-keys';
import { AppView, ConditionalContainer } from '../../common/components';
import Title from './Title';

/**
 * InformationViewProps defines the props for the Information View Component.
 */
interface InformationViewProps {}

/**
 * A component for the Information view
 *
 * @param {InformationViewProps} props - The props for the Information View component.
 */
const InformationView: React.FC<InformationViewProps> = (props: InformationViewProps) => {
  const { t } = useTranslation();
  const { data, loading, error } = useDeviceId();
  return (
    <AppView
      headerMessage={t(TRANSLATION_INFORMATION_KEY)}
      loading={loading}
      error={error}
    >
      <ConditionalContainer display={!!data?.deviceId}>
        <List.Accordion
          title=""
          style={styles.container}
          left={Title}
        >
          <List.Item
            title={t(TRANSLATION_ID_KEY)}
            description={data?.deviceId}
          />
        </List.Accordion>
      </ConditionalContainer>
    </AppView>
  );
};

const styles = StyleSheet.create({
  container: {
    width: '100%',
    maxWidth: 500,
  },
});

export default InformationView;