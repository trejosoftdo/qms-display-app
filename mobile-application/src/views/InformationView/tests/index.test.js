import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import useDeviceId from '../../../hooks/useDeviceId';
import InformationView from '../';


jest.mock('../../../hooks/useDeviceId');

describe('InformationView component', () => {
  const mockData = {
    deviceId: 'mock-device-id',
  };

  afterEach(() => {
    jest.clearAllMocks();
  })

  it('renders the component as expected when loading', () => {
    useDeviceId.mockReturnValue({ loading: true });
    expect(renderShallow(<InformationView />)).toMatchSnapshot();
    expect(useDeviceId).toHaveBeenCalledTimes(1);
    expect(useDeviceId).toHaveBeenCalledWith();
  });

  it('renders the component as expected when data has been loaded', () => {
    useDeviceId.mockReturnValue({
      loading: false,
      data: mockData,
    });
    expect(renderShallow(<InformationView />)).toMatchSnapshot();
    expect(useDeviceId).toHaveBeenCalledTimes(1);
    expect(useDeviceId).toHaveBeenCalledWith();
  });

  it('renders the component as expected when it has errors', () => {
    useDeviceId.mockReturnValue({
      loading: false,
      error: new Error('Unexpected error'),
    });
    expect(renderShallow(<InformationView />)).toMatchSnapshot();
    expect(useDeviceId).toHaveBeenCalledTimes(1);
    expect(useDeviceId).toHaveBeenCalledWith();
  });
});
