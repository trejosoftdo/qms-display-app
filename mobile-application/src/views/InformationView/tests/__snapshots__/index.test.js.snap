// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationView component renders the component as expected when data has been loaded 1`] = `
<AppView
  headerMessage="Translated[translation:information]({})"
  loading={false}
>
  <ConditionalContainer
    display={true}
  >
    <List.Accordion
      left={[Function]}
      style={
        {
          "maxWidth": 500,
          "width": "100%",
        }
      }
      title=""
    >
      <List.Item
        description="mock-device-id"
        title="Translated[translation:id]({})"
      />
    </List.Accordion>
  </ConditionalContainer>
</AppView>
`;

exports[`InformationView component renders the component as expected when it has errors 1`] = `
<AppView
  error={[Error: Unexpected error]}
  headerMessage="Translated[translation:information]({})"
  loading={false}
>
  <ConditionalContainer
    display={false}
  >
    <List.Accordion
      left={[Function]}
      style={
        {
          "maxWidth": 500,
          "width": "100%",
        }
      }
      title=""
    >
      <List.Item
        title="Translated[translation:id]({})"
      />
    </List.Accordion>
  </ConditionalContainer>
</AppView>
`;

exports[`InformationView component renders the component as expected when loading 1`] = `
<AppView
  headerMessage="Translated[translation:information]({})"
  loading={true}
>
  <ConditionalContainer
    display={false}
  >
    <List.Accordion
      left={[Function]}
      style={
        {
          "maxWidth": 500,
          "width": "100%",
        }
      }
      title=""
    >
      <List.Item
        title="Translated[translation:id]({})"
      />
    </List.Accordion>
  </ConditionalContainer>
</AppView>
`;
